src/humanloop/sync/sync_client.py

```diff
@@ -0,0 +1,327 @@
+    manually cleared using the clear_cache() method.
+    """
+    
+    def __init__(
+        self, 
+        client: "BaseHumanloop",
+        base_dir: str = "humanloop",
+        cache_size: int = DEFAULT_CACHE_SIZE,
+        log_level: int = logging.WARNING
+    ):
+        """
+        Parameters
+        ----------
+        client: Humanloop client instance
+        base_dir: Base directory for synced files (default: "humanloop")
+        cache_size: Maximum number of files to cache (default: DEFAULT_CACHE_SIZE)
+        log_level: Log level for logging (default: WARNING)
+        """
+        self.client = client
+        self.base_dir = Path(base_dir)
+        self._cache_size = cache_size
+
+        global logger 
+        logger.setLevel(log_level)
+
+        # Create a new cached version of get_file_content with the specified cache size
+        self.get_file_content = lru_cache(maxsize=cache_size)(self._get_file_content_impl)
+        # Initialize metadata handler
+        self.metadata = MetadataHandler(self.base_dir)
+
+    def _get_file_content_impl(self, path: str, file_type: FileType) -> str:
+        """Implementation of get_file_content without the cache decorator.
+        
+        This is the actual implementation that gets wrapped by lru_cache.
+        
+        Args:
+            path: The normalized path to the file (without extension)
+            file_type: The type of file (Prompt or Agent)
+            
+        Returns:
+            The raw file content
+            
+        Raises:
+            HumanloopRuntimeError: If the file doesn't exist or can't be read
+        """
+        # Construct path to local file
+        local_path = self.base_dir / path
+        # Add appropriate extension
+        local_path = local_path.parent / f"{local_path.stem}.{file_type}"
+        
+        if not local_path.exists():
+            raise HumanloopRuntimeError(f"Local file not found: {local_path}")
+            
+        try:
+            # Read the raw file content
+            with open(local_path) as f:
+                file_content = f.read()
+            logger.debug(f"Using local file content from {local_path}")
+            return file_content
+        except Exception as e:
+            raise HumanloopRuntimeError(f"Error reading local file {local_path}: {str(e)}")
+
+    def get_file_content(self, path: str, file_type: FileType) -> str:
+        """Get the raw file content of a file from cache or filesystem.
+        
+        This method uses an LRU cache to store file contents. When the cache is full,
+        the least recently accessed files are automatically removed to make space.
+        
+        Args:
+            path: The normalized path to the file (without extension)
+            file_type: The type of file (Prompt or Agent)
+            
+        Returns:
+            The raw file content
+            
+        Raises:
+            HumanloopRuntimeError: If the file doesn't exist or can't be read
```

Comment:
```suggestion
            HumanloopRuntimeError: File doesn't exist or the read failed
```

// what does read failed mean? elaborate: disk write errors e.g.